app/models/user.rb

```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
+  validates :name, presence: true,
+            length: {maximum: Settings.users.maximum_name}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true,
+            length: {maximum: Settings.users.maximum_email},
+            format: {with: VALID_EMAIL_REGEX},
+            uniqueness: true
+  has_secure_password
+  validates :password, presence: true,
+            length: {minimum: Settings.users.minimum_password}
```

Comment:
```suggestion
  validates :password, presence: true,
            length: {minimum: Settings.users.minimum_password}
            
  has_secure_password
```
cười rồi vẫn chia tách ng ta là sao cụ

Reply:
em tưởng là có dấu cách hóa ra là em đang chia lìa 2 bạn validates :_(